tracker/track.go

```diff
@@ -152,7 +153,22 @@ func filenameFromTest(t testing.TB) string {
 	if err != nil {
 		t.Fatal(err)
 	}
+	// this is a really hacky way of getting the chain-validation test method name.
+	pc := make([]uintptr, 6)
+	n := runtime.Callers(5, pc)
+	frames := runtime.CallersFrames(pc[:n])
+	frame, _ := frames.Next()
+	frameTokens := strings.Split(frame.Function, ".")
+	parentTestMethodName := frameTokens[2]
+
+	var childTestMethodName string
 	tokens := strings.Split(t.Name(), "/")
-	chainvalTestName := strings.Join(tokens[1:], "")
-	return fmt.Sprintf("/%s", reg.ReplaceAllString(chainvalTestName, ""))
+	// if the parent test doesn't have children
+	if len(tokens) == 1 {
+		childTestMethodName = ""
+	} else {
+		childTestMethodName = tokens[len(tokens)-1]
+	}
+	testFileName := fmt.Sprintf("%s%s", parentTestMethodName, childTestMethodName)
```

Comment:
During normal usage tokens will contain one of the following:

- Parent test has children tests, i.e. calls `t.Run`
```
tokens =
 0 = {string} "TestChainValidationTipSetSuite"                      // test name from go-filecoin or lotus
 1 = {string} "TipSetTest_BlockMessageApplication"                  // (parent) test name defined in chain-validation
 2 = {string} "SECP_and_BLS_messages_cost_different_amounts_of_gas" // (child) test name
```
- Parent test has no children
```
tokens =
 0 = {string} "TestChainValidationMessageSuite"
 1 = {string} "MessageTest_InitActorSequentialIDAddressCreate"
```

During one-off usage tokens will contain one of the following:
- Parent tests has children
```
tokens =
 0 = {string} "TestSomethingRealQuick"
 1 = {string} "success_create_SECP256K1_account_actor"
```
- Parent test has no children
```
tokens = 
 0 = {string} "TestSomethingRealQuick"
```

Walking the stack is the only way (given the current testharness - `*testing.T`) to get the parent tests name.